### ISSUE TITLE
copy_to_tempfile does not work on Windows

### ISSUE THREAD

**Issue by martinsefcik** [janko-m/down#4]
```copy_to_tempfile``` does not work on Windows I think it's because you are trying to move opened IO on line ```FileUtils.mv io.path, tempfile.path```.

```Errno::EACCES: Permission denied @ unlink_internal```

**Comment by janko-m** [comment]
Thanks for reporting, I will push a fix shortly.

**Comment by janko-m** [comment]
Let me know if this indeed fixes the bug, so that I can release a new version with this fix.

**Comment by martinsefcik** [comment]
Still getting the same exception with your fix.

It looks like this is working:
```ruby
    if io.is_a?(OpenURI::Meta) && io.is_a?(Tempfile)
      io.close
      tempfile.close
      FileUtils.mv io.path, tempfile.path
    else
```
tempfile is also opened here and need to be closed before move operation.

**Comment by janko-m** [comment]
Thanks for determining the problem, I released 2.2.1 with the fix.

**Comment by martinsefcik** [comment]
Thanks for fixing and release!
